test/pylib/manager_client.py

```diff
@@ -53,9 +53,12 @@ def __init__(self, sock_path: str, port: int, use_ssl: bool, auth_provider: Any|
         self.api = ScyllaRESTAPIClient()
         self.metrics = ScyllaMetricsClient()
         self.thread_pool = ThreadPoolExecutor()
+        self.test_finished_event = asyncio.Event()
 
     @property
     def client(self):
+        if self.test_finished_event.is_set():
```

Comment:
What if some task first obtains `client`, then goes to sleep, then we call `after_test`, then the task resumes?

Reply:
yes. it is also possible. 
usually, test obtains ```manager``` and use ```client``` as ```manager.client``` and there is an actual workaround for this case

anyway obtaining ```client``` may affect only current current case because all ```clients``` will be closed  in ```manager``` fixture with scope function

as an additional workaround, we can close all clients in after_test->create new during after_test->close again after after_test. But I don't think it is worth it.

Reply:
Let's write a comment about this case, but explain why it's not important / not probable.

I hope we eventually migrate away from the client-rest-server architecture for topology tests, and all of this we'll be able to solve in a much cleaner way.